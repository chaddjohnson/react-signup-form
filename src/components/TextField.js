import React from 'react';
import PropTypes from 'prop-types';
import styled from '@emotion/styled';

const StyledLabel = styled.label`
  display: block;
  margin-bottom: ${props => props.theme.spacing}rem;
`;

const StyledInput = styled.input`
  width: 100%;
  padding: ${props => props.theme.spacing * 2}rem;
  color: ${props => props.theme.colors.text.primary};
  background-color: ${props => props.theme.colors.text.tertiary};
  border-color: ${props =>
    props.error ? props.theme.colors.error : props.theme.colors.text.tertiary};
  border-width: 1px;
  border-style: solid;
  border-radius: 4px;
  font-size: ${props => props.theme.font.size}rem;
`;

const StyledError = styled.div`
  color: ${props => props.theme.colors.error};
  margin-top: ${props => props.theme.spacing}rem;
`;

const TextField = ({
  id,
  name,
  type,
  label,
  placeholder,
  error,
  autoFocus,
  minLength,
  maxLength,
  value,
  onChange,
  onBlur
}) => (
  <>
    <StyledLabel htmlFor={id}>{label}</StyledLabel>
    <StyledInput
      id={id}
      name={name}
      type={type}
      placeholder={placeholder}
      autoFocus={autoFocus}
      minLength={minLength}
      maxLength={maxLength}
      value={value}
      error={error}
      onChange={onChange}
      onBlur={onBlur}
    />
    {error && <StyledError>{error}</StyledError>}
  </>
);

TextField.propTypes = {
  id: PropTypes.string,
  name: PropTypes.string,
  type: PropTypes.oneOf(['text', 'password']),
  label: PropTypes.node,
  placeholder: PropTypes.string,
  error: PropTypes.node,
  autoFocus: PropTypes.bool,
  minLength: PropTypes.number,
  maxLength: PropTypes.number,
  value: PropTypes.string,
  onChange: PropTypes.func,
  onBlur: PropTypes.func
};

TextField.defaultProps = {
  type: 'text',
  autoFocus: false,
  onChange: () => {},
  onBlur: () => {}
};

export default TextField;
