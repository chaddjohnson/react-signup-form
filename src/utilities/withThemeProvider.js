import React from 'react';
import { ThemeProvider } from 'emotion-theming';
import theme from '../theme';

const withThemeProvider = Component => props => (
  <ThemeProvider theme={theme}>
    <Component {...props} />
  </ThemeProvider>
);

export default withThemeProvider;
