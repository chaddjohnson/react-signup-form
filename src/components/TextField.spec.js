import React from 'react';
import renderer from 'react-test-renderer';
import TextField from './TextField';
import withThemeProvider from '../utilities/withThemeProvider';

const ThemedTextField = withThemeProvider(TextField);

describe('TextField', () => {
  it('should render', () => {
    const TextFieldComponent = renderer
      .create(
        <ThemedTextField
          id="test"
          type="text"
          label="Test"
          placeholder="This is a test"
          autoFocus={false}
          minLength={6}
          maxLength={32}
        />
      )
      .toJSON();
    expect(TextFieldComponent).toMatchSnapshot();
  });
});
