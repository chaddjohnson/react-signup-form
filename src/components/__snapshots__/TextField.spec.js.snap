// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TextField should render 1`] = `
Array [
  <label
    className="css-4rywna"
    htmlFor="test"
  >
    Test
  </label>,
  <input
    autoFocus={false}
    className="css-3w2per"
    id="test"
    maxLength={32}
    minLength={6}
    onBlur={[Function]}
    onChange={[Function]}
    placeholder="This is a test"
    type="text"
  />,
]
`;
